tlsconfig/config_test.go

```diff
@@ -12,37 +12,31 @@ import (
 	"testing"
 )
 
-// This is the currently active Let’s Encrypt R3 (RSA 2048, O = Let's Encrypt, CN = R3)
-// cross-signed CA Intermediate cert, downloaded from: https://letsencrypt.org/certs/lets-encrypt-r3-cross-signed.pem
-// It expires Sep 29 19:21:40 2021 GMT
-// download updated versions from https://letsencrypt.org/certificates/
+// This is the currently active Amazon Root CA 1 (CN=Amazon Root CA 1,O=Amazon,C=US),
+// downloaded from: https://www.amazontrust.com/repository/AmazonRootCA1.pem
+// It's valid since May 26 00:00:00 2015 GMT and expires on Jan 17 00:00:00 2038 GMT.
+// Download updated versions from https://www.amazontrust.com/repository/
```

Comment:
Curious; couldn't we use an updated version of the letsencrypt cert? (letsencrypt seems a bit more "independent" than some vendor's root-ca)

Reply:
Initially, I tried updating the intermediate Let's Encrypt R3 cert but it was failing with the error message I documented in commit [tests: Skip tests failing on darwin](https://github.com/docker/go-connections/pull/104/commits/4eb0a2222cf27b5eb084b8a09ec535a8e71175c2).

I wrote a small reproducer to verify the validity of that cert in a Windows 11 VM and it was working properly. Then, I thought that system-wide cert bundle was not up-to-date on Windows and MacOS on GHA runners.


Although it was a dubious theory, I tried to replace that Let's Encrypt cert with the Amazon Root CA 1 cert, but to no avail. Out of laziness (honestly!), I chose to keep this cert but then realized it's valid for a much longer period:

```
$ openssl crl2pkcs7 -nocrl -certfile ~/Downloads/lets-encrypt-r3.pem | openssl pkcs7 -print_certs -text -noout
            Not Before: Sep  4 00:00:00 2020 GMT
            Not After : Sep 15 16:00:00 2025 GMT
$ openssl crl2pkcs7 -nocrl -certfile ~/Downloads/AmazonRootCA1.pem | openssl pkcs7 -print_certs -text -noout
            Not Before: May 26 00:00:00 2015 GMT
            Not After : Jan 17 00:00:00 2038 GMT
```

So I came to the conclusion it's better to keep it.

Also, Let's Encrypt "independence" doesn't mean much in this context. This cert is only used to make sure we can XOR system-wide cert pool and a custom pool.